packages/core-contracts/src/contracts/FleetProxy.sol

```diff
@@ -0,0 +1,262 @@
+import {ProtocolAccessManaged, ContractSpecificRoles} from "@summerfi/access-contracts/contracts/ProtocolAccessManaged.sol";
+import {IFleetCommander} from "../interfaces/IFleetCommander.sol";
+import {IFleetProxy} from "../interfaces/IFleetProxy.sol";
+import {IFleetCommanderConfigProvider} from "../interfaces/IFleetCommanderConfigProvider.sol";
+import {FleetConfig} from "../types/FleetCommanderTypes.sol";
+import {ICrossChainAssetReceiver} from "@summerfi/chain-bridge/interfaces/ICrossChainAssetReceiver.sol";
+
+/**
+ * @title CrossChainFleetProxy
+ * @author SummerFi
+ * @notice Proxy contract that receives and holds assets on a satellite chain on behalf of a source chain fleet
+ * @dev Implements ICrossChainReceiver to handle cross-chain messages
+ */
+contract CrossChainFleetProxy is
+    IFleetProxy,
+    ProtocolAccessManaged,
+    ReentrancyGuard,
+    Pausable,
+    IERC165
+{
+    using SafeERC20 for IERC20;
+
+    /*//////////////////////////////////////////////////////////////
+                            STATE VARIABLES
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice The bridge router used for cross-chain communication
+    IBridgeRouter public immutable bridgeRouter;
+
+    /// @notice The address of the Fleet contract that this proxy covers
+    address public immutable fleetContract;
+
+    /// @notice The bridge options for cross-chain transfers
+    BridgeTypes.BridgeOptions public bridgeOptions;
+
+    /// @notice The address of the source chain's CrossChainArk
+    address public immutable sourceChainArk;
+
+    /*//////////////////////////////////////////////////////////////
+                            EVENTS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Emitted when assets are withdrawn and transferred back to source chain
+    event AssetsWithdrawnAndTransferred(
+        uint256 amount,
+        address asset,
+        uint16 sourceChainId
+    );
+
+    /// @notice Emitted when bridge options are updated
+    event BridgeOptionsUpdated(BridgeTypes.BridgeOptions bridgeOptions);
+
+    /*//////////////////////////////////////////////////////////////
+                            CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Initializes the CrossChainFleetProxy
+     * @param _accessManager Address of the access manager
+     * @param _bridgeRouter Address of the bridge router
+     * @param _fleetContract Address of the Fleet contract this proxy covers
+     * @param _bridgeOptions The bridge options for cross-chain transfers
+     * @param _sourceChainArk Address of the source chain's CrossChainArk
+     */
+    constructor(
+        address _accessManager,
+        address _bridgeRouter,
+        address _fleetContract,
+        BridgeTypes.BridgeOptions memory _bridgeOptions,
+        address _sourceChainArk
+    ) ProtocolAccessManaged(_accessManager) {
```

Comment:
the superchain version, could use peering between crosschain ark(s) and proxi(es)

Reply:
Can't we just keep existing architecture and create an adapter?